python/deltalake/table.py

```diff
@@ -334,9 +334,12 @@ def to_pyarrow_dataset(
             )
         ]
 
-        return FileSystemDataset(
-            fragments, self.schema().to_pyarrow(), format, filesystem
-        )
+        if parquet_read_options is not None and parquet_read_options.dictionary_columns:
+            schema = pyarrow.unify_schemas([f.physical_schema for f in fragments])
```

Comment:
The physical schema doesn't contain the partition columns, so I'm not confident this will work. Perhaps instead we can modify the schema from `schema = self.schema().to_pyarrow()`?

Reply:
You're right, it should be fixed now